src/zarr/storage/common.py

```diff
@@ -66,7 +66,13 @@ async def _init(self, mode: AccessModeLiteral) -> None:
         match mode:
             case "w-":
                 if not await self.empty():
-                    raise FileExistsError(f"{self} must be empty. Mode is 'w-'")
+                    msg = (
+                      f"{self} is not empty, but `mode` is set to 'w-'."
+                      "Either remove the existing objects in storage,"
```

Comment:
I think we need some whitespace: 

```suggestion
                      "Either remove the existing objects in storage, "
```